pandas/core/arrays/categorical.py

```diff
@@ -2159,7 +2167,7 @@ def max(self, skipna=True):
         self.check_for_ordered("max")
 
         if len(self._codes) == 0:
-            return np.nan
+            return na_value_for_dtype(self.dtype)
```

Comment:
this *should* be ```return self.dtype.na_value`` (which can then call this function), BUT this might break some things, I think we have an issue about this @TomAugspurger

Reply:
Probably thinking of https://github.com/pandas-dev/pandas/issues/29962. *Should* be separate from that issue, since we were raising previously rather than returning NaN

Reply:
Hmm, that bug sounds familiar. The datetime categorical test currently fails because `na_value` for categoricals is always `NaN`: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=23025

Reply:
@OliverHofkens you will need to adjust those tests :->